frontend/src/hooks/pims-api/useApiRequestWrapper.ts

```diff
@@ -1,32 +1,44 @@
-import { AxiosError, AxiosPromise } from 'axios';
-import { useCallback, useState } from 'react';
+import axios, { AxiosError, AxiosResponse } from 'axios';
+import useIsMounted from 'hooks/useIsMounted';
+import { IApiError } from 'interfaces/IApiError';
+import { useCallback, useEffect, useState } from 'react';
 import { useDispatch } from 'react-redux';
 import { handleAxiosResponse } from 'utils';
 
 export interface IResponseWrapper<ResponseType> {
   error: AxiosError;
   response: ResponseType;
-  wrappedRequestFunction: () => Promise<ResponseType | undefined>;
+  refresh: (...args: any[]) => Promise<ResponseType | undefined>;
```

Comment:
not a fan of this, may circle back if I have time.

Reply:
would prefer a generic for the request and one for the response maybe? thoughts?

Reply:
for now is fine but yeah I'd prefer a generic for request as well. Much like `React.FC<T>` does for props

Reply:
We can discuss this later today on our dev meeting